stonesoup/metricgenerator/manager.py

```diff
@@ -74,30 +67,65 @@ def generate_metrics(self):
             Metrics generated
         """
 
-        if self.associator is not None and self.association_set is None:
-            self.associate_tracks()
-
         metrics = {}
-        for generator in self.generators:
+
+        generators = self.generators if isinstance(self.generators, list) else [self.generators]
+
+        for generator in generators:
+            if isinstance(generator, SIAPMetrics):
```

Comment:
Referencing to specific metric here doesn't seem appropriate. Why does this need to be done?